test_functions.c

```diff
@@ -0,0 +1,125 @@
+#include "ovector.h"
+#include "test_functions.h"
+#include <stdio.h>
+#include <assert.h>
+
+OVector vector;
+
+OVector vectorDeleted;
+
+void runTests(){
+
+    initVector_conditionsDefault_shouldCreateArray();
+
+    deleteArray__conditionsBad_shouldDeleteArray();
+
+    pushBack__conditionsGood_shouldAddtoArrayInBack();
+
+    pushFront__conditionsGood_shouldAddtoArrayInFront();
+
+    mixPush__conditionsGood_shouldAddElemetns();
+
+    popBack__conditionsGood_shouldRemoveElemetnBack();
+
+    popFront__conditionsGood_shouldRemoveElemetnFront();
+
+    clear__conditionsGood_shouldMakeDefaultOvector();
+
+    stressPush__conditionsGood_shouldAddtoArray();
+
+    removeElem__conditionsGood_shouldRemoveByElem();
+
+    sizeIncrement__conditionsGood_shouldIncreaseSizeOfArray();
+
+    isOvectorNull__conditionsGood_shouldCheckArrayOnNull();
+}
+
+void initVector_conditionsDefault_shouldCreateArray(){
+     initVector(&vector);
+     assert(5 == vector.capacity);
+     ASSERT_EQ(5 , vector.capacity); //??
+}
+
+void deleteArray__conditionsBad_shouldDeleteArray(){
+     initVector(&vectorDeleted);
+     deleteArray(&vectorDeleted);
+     assert(NULL == vectorDeleted.array);
+     initVector(&vectorDeleted);
+     assert(NULL != vectorDeleted.array);
+}
+
+void pushBack__conditionsGood_shouldAddtoArrayInBack(){
+     assert(0 == vector.size);
+     pushBack(&vector, 100);
+     pushBack(&vector, 200);
+     pushBack(&vector, 300);
+     pushBack(&vector, 400);
+     assert(4 == vector.size);
+     assert(400 == vector.array[3]);
```

Comment:
I would also check each vector[x] value it they are correct

`ASSERT_EQ(at(&vector, 0), 100)`
`ASSERT_EQ(at(&vector, 1), 200)`